src/main/java/se/sundsvall/messaging/integration/db/HistoryRepository.java

```diff
@@ -23,4 +27,12 @@ public interface HistoryRepository extends JpaRepository<HistoryEntity, Long>, J
 	List<HistoryEntity> findByMunicipalityIdAndBatchId(String municipalityId, String batchId);
 
 	Page<HistoryEntity> findByMunicipalityIdAndIssuer(String municipalityId, String issuer, Pageable pageable);
+
+	@Query("SELECT DISTINCT h.messageId FROM HistoryEntity h WHERE h.municipalityId = :municipalityId AND h.issuer = :issuer")
```

Comment:
Is this annotation needed, doesn't the methodname `(findDistinctMessageIdsByMunicipalityIdAndIssuer)` already solve this?

Reply:
Since this is selecting just the messageId (a specific field, not the whole entity), the @Query annotation is necessary. Spring Data JPA's query derivation can't handle field selection like that, so yes it is necessary to avoiding the entire object.

Reply:
Can't you use projections?

Reply:
Is there any benefit of using a projection in this case other then to justify a PR comment? We only need distinct messageId's for a specific user and using projection would unnecessarily wrap each string in a separate class/record and require unboxing at a later stage.

Reply:
Here is your projection:
```
public interface MessageId {
   String getMessageId();
}
```

Used as:
`Page<MessageId> findDistinctMessageIdsByMunicipalityIdAndIssuer(...)`

Methods are always better than String querys i.m.h.o, since refactoring, references, etc are easier to find and maintain.
A persistence layer with a lot of string queries will sooner or later be a maintenance hell.